sync-s3.sh

```diff
@@ -52,18 +52,21 @@ while true; do
 done
 
 # Check that the needed programs exist
-for prog in gzip aws; do
+for prog in gzip aws appstream-util; do
     if ! type -p $prog >/dev/null; then
         echo "Cannot find $prog program" >&2
         exit 1
     fi
 done
 
 # Create the eos-extra.xml.gz file if it doesn't exist or is older than
-# eos-extra.xml
+# eos-extra.xml. Validate it first.
 XML="$SRCDIR/app-info/eos-extra.xml"
 XML_GZ="$SRCDIR/s3/app-info/eos-extra.xml.gz"
 XML_GZ_CSUM="${XML_GZ}.sha256sum"
+
+appstream-util validate-strict --nonet "$XML"
```

Comment:
I assume you checked that it passes the strict validation?

Reply:
It does pass that validation for me locally (with this PR applied).

Reply:
I also tested that it fails validation if I deliberately break the XML in a few different ways.

Reply:
In my experience the strict validation can be a bit too strict. That's why the eos-build code uses `validate-relax`, but we can see how it goes. FWIW, the version of `appstream-glib` on the builders is https://obs-master.endlessm-sf.com/package/show/infra:xenial/appstream-glib, which is pretty old now. FWIW, I'll probably update the builders to debian buster (when it's released) or ubuntu bionic. Those have 0.7.14 and 0.7.7, respectively.